src/Domain/CoordinateMaps/BulgedCube.cpp

```diff
@@ -0,0 +1,270 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/BulgedCube.hpp"
+
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace CoordinateMaps {
+BulgedCube::BulgedCube(const double radius, const double sphericity,
+                       const bool with_equiangular_map)
+    : radius_(radius),
+      sphericity_(sphericity),
+      with_equiangular_map_(with_equiangular_map) {
+  ASSERT(radius > 0, "The radius of the cube must be greater than zero");
+}
+
+template <typename T>
+std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3> BulgedCube::
+operator()(const std::array<T, 3>& x) const noexcept {
+  using ReturnTypeElement = std::decay_t<tt::remove_reference_wrapper_t<T>>;
+  const ReturnTypeElement& xi = x[0];
+  const ReturnTypeElement& eta = x[1];
+  const ReturnTypeElement& zeta = x[2];
+  const ReturnTypeElement cap_xi =
+      with_equiangular_map_ ? tan(M_PI_4 * static_cast<ReturnTypeElement>(x[0]))
+                            : static_cast<ReturnTypeElement>(x[0]);
+  const ReturnTypeElement cap_eta =
+      with_equiangular_map_ ? tan(M_PI_4 * static_cast<ReturnTypeElement>(x[1]))
+                            : static_cast<ReturnTypeElement>(x[1]);
+  const ReturnTypeElement cap_zeta =
+      with_equiangular_map_ ? tan(M_PI_4 * static_cast<ReturnTypeElement>(x[2]))
+                            : static_cast<ReturnTypeElement>(x[2]);
+
+  const double flat_face_factor = (1.0 - sphericity_) * radius_ / sqrt(3.0);
+  const double sphericity_times_radius = sphericity_ * radius_;
+  const double spherical_face_factor = (sphericity_times_radius) / sqrt(3.0);
+  const ReturnTypeElement one_over_rho_xi = 1.0 / sqrt(2.0 + square(cap_xi));
+  const ReturnTypeElement one_over_rho_eta = 1.0 / sqrt(2.0 + square(cap_eta));
+  const ReturnTypeElement one_over_rho_zeta =
+      1.0 / sqrt(2.0 + square(cap_zeta));
+  const ReturnTypeElement one_over_rho_xi_eta =
+      1.0 / sqrt(1.0 + square(cap_xi) + square(cap_eta));
+  const ReturnTypeElement one_over_rho_xi_zeta =
+      1.0 / sqrt(1.0 + square(cap_xi) + square(cap_zeta));
+  const ReturnTypeElement one_over_rho_eta_zeta =
+      1.0 / sqrt(1.0 + square(cap_eta) + square(cap_zeta));
+
+  const ReturnTypeElement physical_x =
```

Comment:
Why are these const? Now you have to copy them again when you return them, instead of being able to move them into the array.